web/src/lib/stores/assets-store.svelte.ts

```diff
@@ -1292,12 +1324,91 @@ export class AssetStore {
     return this.buckets[0]?.getFirstAsset();
   }
 
-  async getPreviousAsset(asset: AssetResponseDto): Promise<AssetResponseDto | undefined> {
-    let bucket = await this.#getBucketInfoForAsset(asset);
+  async getPreviousAsset(
+    idable: { id: string },
+    skipTo: 'asset' | 'day' | 'month' | 'year' = 'asset',
+  ): Promise<AssetResponseDto | undefined> {
+    const bucket = this.#findBucketForAsset(idable.id);
     if (!bucket) {
       return;
     }
+    const asset = bucket.findById(idable.id);
```

Comment:
I was quite confused here because I expected this function to return a bucket (by id). Could we rename this to `findAssetById` instead?